drivers/flash/nrf_qspi_nor.c

```diff
@@ -556,11 +587,28 @@ static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 		return -EINVAL;
 	}
 
-	qspi_lock(dev);
 
-	nrfx_err_t res = nrfx_qspi_write(src, size, addr);
+	nrfx_err_t res;
 
-	qspi_wait_for_completion(dev, res);
+	if (IS_ENABLED(CONFIG_NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH) &&
```

Comment:
Shouldn't this fail when the config is not enabled and the source is in FLASH? Now it will go into the `else`.

Reply:
that's fine, then the NRFX driver will fail and the error is reported back to the user.

Reply:
Ok, that's not quite clear to the. I still think the config is not needed.

You said:

> This allows the user to either use this workaround or copy their data in RAM in a local function doing it more efficiently.

They can still do that if there's no config. If they use a RAM buffer, this buffer will not be used.